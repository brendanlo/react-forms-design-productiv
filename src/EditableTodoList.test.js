import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

describe("editable todo list", function () {

  const testTodoList = [{   id: 1,
                            title: "test1", 
                            description: "testing1", 
                            priority: 1 },
                        {   id: 2,
                            title: "test2",
                            description: "testing2",
                            priority: 2 }
                        ];

  it("renders without crashing", function () {
    render(<EditableTodoList todos={testTodoList} />);
  });

  it("contains all todos", function () {
    const result = render(<EditableTodoList todos={testTodoList} />);
    expect(result.queryByText("test1")).toBeInTheDocument();
    expect(result.queryByText("test2")).toBeInTheDocument();
    expect(result.queryByText("testing2")).toBeInTheDocument();
  });

  it("Snapshot", function () {
    const result = render(<EditableTodoList todos = {testTodoList} />);
    expect(result).toMatchSnapshot();
  })
});
