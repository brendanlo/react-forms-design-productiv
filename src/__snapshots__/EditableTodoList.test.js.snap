// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editable todo list Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="EditableTodoList"
      >
        <div
          class="EditableTodo"
        >
          <div
            class="mb-3"
          >
            <div
              class="float-end text-sm-end"
            >
              <button
                class="EditableTodo-toggle btn-link btn btn-sm"
              >
                Edit
              </button>
              <button
                class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              >
                Del
              </button>
            </div>
            <div
              class="Todo"
            >
              <div
                class="Todo-title"
              >
                <b>
                  test1
                </b>
                <small>
                  (priority: 
                  1
                  )
                </small>
              </div>
              <div
                class="Todo-description"
              >
                <small>
                  testing1
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          class="EditableTodo"
        >
          <div
            class="mb-3"
          >
            <div
              class="float-end text-sm-end"
            >
              <button
                class="EditableTodo-toggle btn-link btn btn-sm"
              >
                Edit
              </button>
              <button
                class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              >
                Del
              </button>
            </div>
            <div
              class="Todo"
            >
              <div
                class="Todo-title"
              >
                <b>
                  test2
                </b>
                <small>
                  (priority: 
                  2
                  )
                </small>
              </div>
              <div
                class="Todo-description"
              >
                <small>
                  testing2
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="EditableTodoList"
    >
      <div
        class="EditableTodo"
      >
        <div
          class="mb-3"
        >
          <div
            class="float-end text-sm-end"
          >
            <button
              class="EditableTodo-toggle btn-link btn btn-sm"
            >
              Edit
            </button>
            <button
              class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            >
              Del
            </button>
          </div>
          <div
            class="Todo"
          >
            <div
              class="Todo-title"
            >
              <b>
                test1
              </b>
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div
              class="Todo-description"
            >
              <small>
                testing1
              </small>
            </div>
          </div>
        </div>
      </div>
      <div
        class="EditableTodo"
      >
        <div
          class="mb-3"
        >
          <div
            class="float-end text-sm-end"
          >
            <button
              class="EditableTodo-toggle btn-link btn btn-sm"
            >
              Edit
            </button>
            <button
              class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            >
              Del
            </button>
          </div>
          <div
            class="Todo"
          >
            <div
              class="Todo-title"
            >
              <b>
                test2
              </b>
              <small>
                (priority: 
                2
                )
              </small>
            </div>
            <div
              class="Todo-description"
            >
              <small>
                testing2
              </small>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
